src/backend/distributed/commands/function.c

```diff
@@ -187,22 +187,62 @@ create_distributed_function(PG_FUNCTION_ARGS)
 	EnsureFunctionOwner(funcOid);
 
 	ObjectAddressSet(functionAddress, ProcedureRelationId, funcOid);
-	ErrorIfFunctionDependsOnExtension(&functionAddress);
 
 	/*
-	 * when we allow propagation within a transaction block we should make sure to only
-	 * allow this in sequential mode
+	 * If the function is owned by an extension, only update the
+	 * pg_dist_object, and not propagate the CREATE FUNCTION. Function
+	 * will be created by the virtue of the extension creation.
 	 */
-	EnsureSequentialModeForFunctionDDL();
+	if (IsObjectAddressOwnedByExtension(&functionAddress, &extensionAddress))
+	{
+		if (IsObjectDistributed(&functionAddress))
+		{
+			ereport(ERROR, (errmsg("Function(%s) is already distributed.",
+								   get_func_name(funcOid))));
+		}
+
+		/*
+		 * Ensure corresponding extension is in pg_dist_object.
+		 */
+
+		Oid citusExtensionOid = get_extension_oid("citus", false);
+
+		if (citusExtensionOid == extensionAddress.objectId)
+		{
+			/* Citus extension is a special case */
+			ereport(ERROR, (errmsg("Citus extension functions(%s) "
+								   "can't be distributed.",
+								   get_func_name(funcOid))));
+		}
+		else if (!IsObjectDistributed(&extensionAddress))
+		{
+			ereport(WARNING, (errmsg("Extension(%s) owning the "
```

Comment:
I'm not sure we should have a WARNING here. We routinely call EnsureDependenciesExistOnAllNodes in different scenarios. Should this be a DEBUG1 for the test?

Reply:
Agree with Marco, we definitely don't warn on this, as Citus should simply resolve the distribution transparently.

Reply:
Changed to DEBUG1